lib/client.js

```diff
@@ -159,11 +164,6 @@ var Client = function(options) {
   this._context = options.context || null;
   this.ready = false;
 
-  if (this._parent) {
-    this.ready = this._parent.ready;
-    return this; // shortcut handshake
-  }
-
   this.on('app.registered', function(data) {
```

Comment:
Yea, I think it should fire in time if your app itself is creating the instance (i.e. modals). But if it doesn't it shouldn't matter, the parent app can still retrieve the context using the `instances` or `context` API from the instance client.